templates/server/server_ssl_settings.erb

```diff
@@ -26,7 +26,9 @@
 <% end -%>
   ssl_protocols             <%= @ssl_protocols %>;
   ssl_ciphers               <%= @ssl_ciphers %>;
-  ssl_prefer_server_ciphers on;
+<%- if @ssl_prefer_server_ciphers -%>
```

Comment:
I'm not up to date on Puppet 4 syntax. But if Enum only allows `yes` or `no` why have a conditional in the template? It will always be either "yes" or "no" and with that logic the conditional will always be true, so the directive will always end up in the generated config.